spi/common/aggregate-service-spi/src/main/java/org/eclipse/edc/service/spi/result/ServiceResult.java

```diff
@@ -57,7 +57,20 @@ public static <T> ServiceResult<T> from(StoreResult<T> storeResult) {
         if (storeResult.succeeded()) {
             return success(storeResult.getContent());
         }
+        switch (storeResult.reason()) {
+            case NOT_FOUND:
+                return notFound(storeResult.getFailureDetail());
+            case ALREADY_EXISTS:
+                return conflict(storeResult.getFailureDetail());
+            default:
+                return badRequest(storeResult.getFailureDetail());
+        }
+    }
 
+    public static <T> ServiceResult<T> fromFailure(StoreResult<?> storeResult) {
+        if (storeResult.succeeded()) {
+            throw new IllegalArgumentException("Can only use fromFailure() when the argument is a failed result. Please use ServiceResult.from() to also convert success results!");
```

Comment:
Can you not reference fromFailure() b/c of potential refactoring issues, e.g. spell it out or something else.